Automattic/pocket-casts-android
# End of Year: Top podcast and longest episode stories design

| 📘 Project: #410 |
|:---:|

iOS PRs: 
https://github.com/Automattic/pocket-casts-ios/pull/455
https://github.com/Automattic/pocket-casts-ios/pull/463

## Description
Add the design for the top podcast and longest episode stories which were similar in design.

The layers of the displayed podcast and episode are achieved by using the podcast cover colors informed by the server.

PS: I also extracted a few components and modifiers (e.g. `PodcastCover`, `Modifier.podcastDynamicBackground`). This has impacted previous stories and increased no. of lines in this PR.

## Testing Instructions

1. Set `END_OF_YEAR_ENABLED` feature flag to `true` in `base.gradle`
2. Run the app and make sure you're logged in to an account with a few items on Listening History
3. When the prompt appears, tap "View My 2022"
4. Navigate to the top podcast story
5. ✅ Check the design
6. Navigate to the longest episode story
7. ✅ Check the design

## Screenshots or Screencast 

|Top Podcast (Pixel 2) | Longest Episode (Pixel 5) |
|------|----|
| <img with = 320 src="https://user-images.githubusercontent.com/1405144/201283586-1fc5f9d1-89ee-443c-bbf0-840f7aa2b97d.png"/>  | <img with = 320 src="https://user-images.githubusercontent.com/1405144/201282335-e14b9c16-4b6b-4c0f-8610-62c511c1f82c.png"/>|

## Checklist
- If this is a user-facing change, I have added an entry in CHANGELOG.md
- I have considered whether it makes sense to add tests for my changes
 
#### I have tested any UI changes...

- with different themes
- with a landscape orientation
- with the device set to have a large display and font size
